pkg/types/installconfig.go

```diff
@@ -133,12 +132,16 @@ type Networking struct {
 	// Default is a single cluster network with a CIDR of 10.128.0.0/14
 	// and a host subnet length of 9. The default is only applicable if PodCIDR
 	// is not present.
-	ClusterNetworks []netopv1.ClusterNetwork `json:"clusterNetworks,omitempty"`
+	ClusterNetworks []ClusterNetworkEntry `json:"clusterNetworks,omitempty"`
+}
 
-	// PodCIDR is deprecated (and badly named; it should have always
-	// been called ClusterCIDR. If no ClusterNetworks are specified,
-	// we will fall back to the PodCIDR
-	// TODO(cdc) remove this.
-	// +optional
-	PodCIDR *ipnet.IPNet `json:"podCIDR,omitempty"`
+// ClusterNetworkEntry is a single IP address block for pod IP blocks. IP blocks
+// are allocated with size 2^HostSubnetLength.
+type ClusterNetworkEntry struct {
+	// The IP block address pool
+	CIDR ipnet.IPNet `json:"cidr"`
+
+	// The size of blocks to allocate from the larger pool.
+	// This is the length in bits - so a 9 here will allocate a /23.
+	HostSubnetLength uint32 `json:"hostSubnetLength"`
```

Comment:
Can we make this a signed integer, with validation, to conform to kube API conventions?

https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#primitive-types

Reply:
done